internal/pkg/vault/request.go

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ * Copyright 2021 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell / Alain Pulluelo, ForgeRock AS
+ *******************************************************************************/
+
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients"
+)
+
+// parameters structure for request method
+type RequestArgs struct {
+	// Authentication token
+	AuthToken string
+	// HTTP method
+	Method string
+	// URL path
+	Path string
+	// If non-nil, passed to JSON serializer and included in request
+	JSONObject interface{}
+	// Included in HTTP request if JSONObject is nil
+	BodyReader io.Reader
+	// Description of the operation being performed included in log messages
+	OperationDescription string
+	// Expected status code to be returned from HTTP request
+	ExpectedStatusCode int
```

Comment:
I need to understand the background of this assumption more: Not sure this assumption is ok: in some cases, there may be more than one expected status codes from the client. One case is like `unseal` the Vault, 200 might not be the only expected code from the client point of view; maybe all status code in the range of 200-299 would be ok?

Reply:
This is the existing code from SecretStoreClient that is working. It is making a specific request with a specific known status code expected.

Reply:
ok, now I see this better: only applies to a single request in a http url call.